pkg/handler/enqueue.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var enqueueLog = logf.KBLog.WithName("eventhandler").WithName("Enqueue")
+
+var _ EventHandler = &Enqueue{}
+
+// Enqueue enqueues a Request containing the Name and Namespace of the object for each event.
+type Enqueue struct{}
+
+// Create implements EventHandler
+func (e *Enqueue) Create(q workqueue.RateLimitingInterface, evt event.CreateEvent) {
+	if evt.Meta == nil {
```

Comment:
:-/ I think we want to Inject a logger if we can, so that we can track *which* enqueue is having issues.